Remotion/Validation/UnitTests/Rules/PropertyValidationRuleTest.cs

```diff
@@ -1,33 +1,242 @@
+    {
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      var propertyFailure = new PropertyValidationFailure(new Person(), propertyInformationMock.Object, "propval", "err", "validmessage");
+
+      var validatorMock = new Mock<IPropertyValidator>();
+
+      PropertyValidatorContext actualPropertyValidatorContext = null;
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Callback((PropertyValidatorContext ctx) => { actualPropertyValidatorContext = ctx; })
+          .Returns(new[] { propertyFailure });
+
+
+      object propertyObj = null;
+      Func<object, object> propertyFunc = obj =>
+      {
+        propertyObj = obj;
+        return "test";
+      };
+
+      object conditionObj = null;
+      Func<object, bool> conditionFunc = obj =>
+      {
+        conditionObj = obj;
+        return true;
+      };
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, conditionFunc, new[] { validatorMock.Object });
+
+      var context = new ValidationContext(new Person());
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.EqualTo(new[] { propertyFailure }));
+      Assert.That(context.InstanceToValidate, Is.SameAs(actualPropertyValidatorContext.Instance));
+      Assert.That(context.InstanceToValidate, Is.SameAs(conditionObj));
+      Assert.That(context.InstanceToValidate, Is.SameAs(propertyObj));
+      Assert.That("test", Is.EqualTo(actualPropertyValidatorContext.PropertyValue));
+    }
+
+    [Test]
+    public void Validate_WithoutCondition_ReturnsValidationResults ()
+    {
+      var propertyInformationMock = new Mock<IPropertyInformation>();
+
+      var propertyFailure = new PropertyValidationFailure(new Person(), propertyInformationMock.Object, "propval", "err", "validmessage");
+
+      var validatorMock = new Mock<IPropertyValidator>();
+
+      PropertyValidatorContext actualPropertyValidatorContext = null;
+      validatorMock
+          .Setup(_ => _.Validate(It.IsAny<PropertyValidatorContext>()))
+          .Callback((PropertyValidatorContext ctx) => { actualPropertyValidatorContext = ctx; })
+          .Returns(new[] { propertyFailure });
+
+      object propertyObj = null;
+      Func<object, object> propertyFunc = obj =>
+      {
+        propertyObj = obj;
+        return "test";
+      };
+
+      var rule = new PropertyValidationRule<object, int>(propertyInformationMock.Object, propertyFunc, null, new[] { validatorMock.Object });
+
+      var context = new ValidationContext(new Person());
+      var result = rule.Validate(context);
+
+      Assert.That(result, Is.EqualTo(new[] { propertyFailure }));
+      Assert.That(context.InstanceToValidate, Is.SameAs(actualPropertyValidatorContext.Instance));
+      Assert.That(context.InstanceToValidate, Is.SameAs(propertyObj));
+      Assert.That("test", Is.EqualTo(actualPropertyValidatorContext.PropertyValue));
```

Comment:
please check. you flipped actual and expected. recommend you add `actual` (or `result`, depending on context) to each variable name.